packages/frontend/web-ui/src/auth/hooks/useRegisterConfirm.spec.js

```diff
@@ -0,0 +1,281 @@
+import { afterAll, beforeAll, describe, expect, it, jest } from '@jest/globals';
+import { act, renderHook } from '@testing-library/react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  STATUS_FULFILLED,
+  STATUS_REJECTED,
+  UNEXPECTED_ERROR_MESSAGE,
+  useRegisterConfirm,
+} from './useRegisterConfirm';
+import { httpClient } from '../../common/http/services/HttpService';
+import { buildSerializableResponse } from '../../common/http/helpers';
+
+jest.mock('react-redux', () => {
+  return {
+    ...jest.requireActual('react-redux'),
+    useDispatch: jest.fn(),
+    useSelector: jest.fn(),
+  };
+});
+jest.mock('../../common/http/services/HttpService');
+jest.mock('../../common/http/helpers');
+
+describe(useRegisterConfirm.name, () => {
+  let authErrorMessageFixture;
+  let dispatchMock;
+  let tokenFixture;
+
+  beforeAll(() => {
+    authErrorMessageFixture = null;
+    dispatchMock = jest.fn().mockResolvedValue(undefined);
+    tokenFixture = null;
+  });
+
+  describe('having a window with location.href with code query', () => {
+    let previousLocation;
+    let locationFixture;
+
+    beforeAll(() => {
+      previousLocation = window.location;
+      locationFixture = new URL('http://corniegame.com/auth/path?code=code');
+
+      Object.defineProperty(window, 'location', {
+        value: new URL(locationFixture),
+        configurable: true,
+      });
+    });
+
+    describe('when called, and httpClient.updateUserMe() returns an OK response', () => {
+      let result;
+      let status;
+      let errorMessage;
+
+      beforeAll(async () => {
+        tokenFixture = 'jwt token fixture';
+
+        useDispatch.mockReturnValue(dispatchMock);
+        useSelector.mockImplementation(() => ({
+          token: tokenFixture,
+          errorMessage: authErrorMessageFixture,
+        }));
+
+        httpClient.updateUserMe.mockReturnValueOnce({
+          headers: {},
+          body: {
+            active: true,
+            id: 'id',
+            name: 'name',
+          },
+          statusCode: 200,
+        });
+
+        buildSerializableResponse.mockImplementation((response) => ({
+          body: response.body,
+          statusCode: response.statusCode,
+        }));
+
+        await act(() => {
+          result = renderHook(() => useRegisterConfirm()).result;
+        });
+
+        status = result.current.status;
+        errorMessage = result.current.errorMessage;
+      });
+
+      afterAll(() => {
+        jest.resetAllMocks();
+        jest.clearAllMocks();
+      });
+
+      it('should called useDispatch()', () => {
+        expect(useDispatch).toHaveBeenCalled();
```

Comment:
`toHaveBeenCalledTimes` is more accurate.

In addition, assert mock arguments with `toHaveBeenCalledWith`.

Reply:
With `toHaveBeenCalledTimes` was called three times because the component does 3 iterations until it is rendered. I have done the `toHaveBeenCalledWith`.